youtube_dl/extractor/vidlii.py

```diff
@@ -46,9 +67,31 @@ def _real_extract(self, url):
         webpage = self._download_webpage(
             'https://www.vidlii.com/watch?v=%s' % video_id, video_id)
 
-        video_url = self._search_regex(
-            r'src\s*:\s*(["\'])(?P<url>(?:https?://)?(?:(?!\1).)+)\1', webpage,
-            'video url', group='url')
+        formats = []
+
+        def add_format(format_url, height=None):
+            height = int(self._search_regex(r"(\d+).mp4",
+                         format_url, "height", default=360))
+
+            formats.append({
+                'url': format_url,
+                'format_id': '%dp' % height if height else None,
+                'height': height,
+            })
+
+        sources = re.findall(
+            r'src\s*:\s*(["\'])(?P<url>(?:https?://)?(?:(?!\1).)+)\1',
+            webpage)
+
+        try:
+            self._request_webpage(sources[1][1], video_id, 'Checking HD URL')
+            add_format(sources[1][1])
+        except ExtractorError:
+            pass
+
+        add_format(sources[0][1])
+
+        self._sort_formats(formats)
 
         title = self._search_regex(
```

Comment:
Not part of your PR, but should convert HTML-encoding to text, also strip surrounding spaces:
```suggestion
        title = self._html_search_regex(
```